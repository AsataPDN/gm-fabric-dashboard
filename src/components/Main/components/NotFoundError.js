import React from "react";
import { PropTypes } from "prop-types";

import ErrorBox from "./ErrorBox";
import ErrorContent from "./ErrorContent";
import Span from "./Span";

import Icon from "components/Icon";
import Glyph from "components/Glyphs/index";

/**Stateless functional React component that renders the error message box
 * Takes an error message and returns error message box
 * @param {string} errorMsg
 * @returns react component
 */

const NotFoundError = ({ errorMsg = "Not Found" }) => {
  return (
    <ErrorBox>
      <ErrorContent>
        <Span>
          <Icon borderStyle="BorderTriangleSmall" iconRatio={"3"}>
            <Glyph name={"Exclamation"} />
          </Icon>
        </Span>
        <Span>{errorMsg}</Span>
      </ErrorContent>
    </ErrorBox>
  );
};

NotFoundError.propTypes = {
  errorMsg: PropTypes.string
};

export default NotFoundError;
