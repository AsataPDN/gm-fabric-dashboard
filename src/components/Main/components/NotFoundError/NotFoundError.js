import React from "react";
import { PropTypes } from "prop-types";

import ErrorBox from "./components/ErrorBox";
import ErrorContent from "./components/ErrorContent";

import Icon from "components/Icon";
import Glyph from "components/Glyphs/index";
import Span from "components/Main/components/Span";

/**Stateless functional React component that renders the error message box
 * Takes an error message and returns error message box
 * @param {string} errorMsg
 * @returns react component
 */

const NotFoundError = ({ errorMsg = "Not Found" }) => {
  return (
    <ErrorBox>
      <ErrorContent>
        <Span>
          <Icon borderStyle="BorderTriangleSmall" iconRatio={"3"}>
            <Glyph name={"Exclamation"} />
          </Icon>
        </Span>
        <Span>{errorMsg}</Span>
      </ErrorContent>
    </ErrorBox>
  );
};

NotFoundError.propTypes = {
  errorMsg: PropTypes.string
};

export default NotFoundError;
