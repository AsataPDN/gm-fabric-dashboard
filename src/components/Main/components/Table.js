import styled from "styled-components";

import { FONT_STACK_DATA } from "../../../style/styleVariables";

const Table = styled.div`
  width: 100%;
  font-family: ${FONT_STACK_DATA};
`;

export default Table;
