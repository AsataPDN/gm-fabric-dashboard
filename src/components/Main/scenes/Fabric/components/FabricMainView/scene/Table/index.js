import React from "react";
import { PropTypes } from "prop-types";
import _ from "lodash";
import { microserviceStatuses } from "utils/constants";

import styled from "styled-components";

import GMServiceList from "./components/GMServiceList";
import GMServiceHeader from "../../components/GMServiceHeader";
import {
  COLOR_CONTENT_BACKGROUND,
  ZINDEX_STICKY,
  CONTENT_MAX_WIDTH
} from "style/styleVariables";
import { edgeColor, spacingScale } from "style/styleFunctions";
import { withRouter } from "react-router-dom";
// styled components
const SectionContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${spacingScale(1)} 0 0;
  border-top: 1px solid ${edgeColor(COLOR_CONTENT_BACKGROUND).string()};
  flex-wrap: nowrap;

  &:first-of-type {
    border-top: 0;
  }
`;

const SectionHeader = styled.div`
  width: 100%;
  display: flex;
  margin: 0 auto;
  padding: 0 ${spacingScale(2)};
  position: sticky;
  top: 0;
  z-index: ${ZINDEX_STICKY};
  background-color: ${COLOR_CONTENT_BACKGROUND.string()};
  max-width: ${CONTENT_MAX_WIDTH};
`;

const SectionContent = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0 0 ${spacingScale(1)};
  max-width: ${CONTENT_MAX_WIDTH};
  width: 100%;
  margin-left: auto;
  margin-right: auto;
`;

GMServiceListView.propTypes = {
  groupByAttribute: PropTypes.string.isRequired,
  location: PropTypes.object,
  services: PropTypes.array.isRequired,
  sortByAttribute: PropTypes.string.isRequired
};

function GMServiceListView({
  groupByAttribute,
  sortByAttribute,
  services,
  location: { pathname }
}) {
  // get unique headers
  if (groupByAttribute !== "None") {
    const dataGroupedByHeader = _.groupBy(services, item =>
      item.headerTitle.toLowerCase()
    );
    const headerTitles = Object.keys(dataGroupedByHeader);
    const statusSort = pathname.slice(1);

    let headers;

    // If we pulled a status from the path, use that as the header
    // Else if we are on the main fabric grid, use microServiceStatuses as the headers
    if (statusSort && groupByAttribute === "Status") {
      headers = [statusSort];
    } else if (!statusSort && groupByAttribute === "Status") {
      headers = microserviceStatuses.map(item => item.toLowerCase());
    } else {
      headers = headerTitles;
    }

    return (
      <div>
        {headers.map((header, i) => (
          <SectionContainer key={header}>
            <SectionHeader>
              <GMServiceHeader
                headerTitle={header}
                showStatusIcon={groupByAttribute === "Status"}
              />
            </SectionHeader>
            <SectionContent>
              <GMServiceList
                items={_.orderBy(
                  dataGroupedByHeader[header],
                  [sortByAttribute.toLowerCase(), "name"],
                  ["asc", "asc"]
                )}
                groupByAttribute={groupByAttribute}
              />
            </SectionContent>
          </SectionContainer>
        ))}
      </div>
    );
  } else {
    return (
      <div>
        <SectionContainer>
          <SectionContent>
            <GMServiceList
              items={_.orderBy(
                services,
                [sortByAttribute.toLowerCase(), "name"],
                ["asc", "asc"]
              )}
            />
          </SectionContent>
        </SectionContainer>
      </div>
    );
  }
}

export default withRouter(GMServiceListView);
