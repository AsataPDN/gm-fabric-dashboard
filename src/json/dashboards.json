{
  "http": {
    "name": "HTTP",
    "runtime": "JVM",
    "summaryCard": {
      "lineOne": [
        {
          "type": "string",
          "value": "HTTP:"
        },
        {
          "type": "latest",
          "value": "http/success"
        },
        {
          "type": "string",
          "value": "/"
        },
        {
          "type": "latest",
          "value": "http/requests"
        }
      ],
      "lineTwo": [
        {
          "type": "string",
          "value": "HTTPS:"
        },
        {
          "type": "latest",
          "value": "https/success"
        },
        {
          "type": "string",
          "value": "/"
        },
        {
          "type": "latest",
          "value": "https/requests"
        }
      ]
    },
    "grid": {
      "breakpoints": {
        "lg": 1200,
        "md": 996,
        "sm": 768
      },
      "cols": {
        "lg": 12,
        "md": 8,
        "sm": 4
      },
      "layouts": {
        "lg": [
          { "i": "HTTP Connections",    "x": 0, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "Data Transfer Rates", "x": 4, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "HTTP Stats",          "x": 8, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 }
        ],
        "md": [
          { "i": "HTTP Connections",    "x": 0, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "Data Transfer Rates", "x": 4, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "HTTP Stats",          "x": 8, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 }
        ],
        "sm": [
          { "i": "HTTP Connections",    "x": 0, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "Data Transfer Rates", "x": 4, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 },
          { "i": "HTTP Stats",          "x": 8, "y": 0, "w": 4, "h": 6, "minW": 3, "minH": 4 }
        ]
      },
      "rowHeight": 60
    },
    "charts": [
      {
        "title": "HTTP Connections",
        "type": "GMLineChart",
        "data": {
          "timeseries": [
            {
              "type": "value",
              "attribute": "http/connections",
              "label": "HTTP Connections"
            }
          ]
        }
      },
      {
        "title": "Data Transfer Rates",
        "type": "GMLineChart",
        "data": {
          "timeseries": [
            {
              "type": "netChange",
              "attribute": "http/sent_bytes",
              "label": "HTTP Sent Bytes (kBps)",
              "baseUnit": "B",
              "resultUnit": "kB",
              "precision": 3
            },
            {
              "type": "netChange",
              "attribute": "https/sent_bytes",
              "label": "HTTPS Sent Bytes (kBps)",
              "baseUnit": "B",
              "resultUnit": "kB",
              "precision": 3
            },
            {
              "type": "netChange",
              "attribute": "http/received_bytes",
              "label": "HTTP Received (kBps)",
              "baseUnit": "B",
              "resultUnit": "kB",
              "precision": 3
            },
            {
              "type": "netChange",
              "attribute": "https/received_bytes",
              "label": "HTTPS Received (kBps)",
              "baseUnit": "B",
              "resultUnit": "kB",
              "precision": 3
            }
          ]
        }
      },
      {
        "title": "HTTP Stats",
        "type": "GMTable",
        "data": {
          "headers": [
            "Requests",
            "Success"
          ],
          "rows": [
            [
              "http",
              "http/requests",
              "http/success"
            ],
            [
              "https",
              "https/requests",
              "https/success"
            ]
          ]
        }
      }
    ]
  },
  "jvm": {
    "name": "JVM",
    "runtime": "JVM",
    "summaryCard": {
      "chart": {
        "type": "value",
        "dataAttribute": "jvm/mem/current/used"
      },
      "lineOne": [
        {
          "type": "latest",
          "value": "jvm/mem/current/used",
          "baseUnit": "B",
          "resultUnit": "MB",
          "precision": 3
        },
        {
          "type": "string",
          "value": "MB"
        }
      ],
      "lineTwo": []
    },
    "grid": {
      "breakpoints": {
        "lg": 1200,
        "md": 996,
        "sm": 768
      },
      "cols": {
        "lg": 12,
        "md": 8,
        "sm": 4
      },
      "layouts": {
        "lg": [
          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
        ],
        "md": [
          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
        ],
        "sm": [
          { "i": "Heap",                "x": 0, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 },
          { "i": "Classes",             "x": 6, "y": 0, "w": 6, "h": 9, "minW": 4, "minH": 5 }
        ]
      },
      "rowHeight": 60
    },
    "charts": [
      {
        "title": "Heap",
        "type": "GMLineChart",
        "data": {
          "detailLines": [
            [
              {
                "type": "string",
                "value": "Max Heap:"
              },
              {
                "type": "latest",
                "value": "jvm/heap/max",
                "baseUnit": "B",
                "resultUnit": "GB",
                "precision": 3
              },
              {
                "type": "string",
                "value": "GB"
              }
            ]
          ],
          "timeseries": [
            {
              "type": "value",
              "attribute": "jvm/heap/committed",
              "label": "JVM Heap Committed (MB)",
              "baseUnit": "B",
              "resultUnit": "MB",
              "precision": 3
            },
            {
              "type": "value",
              "attribute": "jvm/heap/used",
              "label": "JVM Heap Used (MB)",
              "baseUnit": "B",
              "resultUnit": "MB",
              "precision": 3
            }
          ]
        }
      },
      {
        "title": "Classes",
        "type": "GMLineChart",
        "data": {
          "detailLines": [
            [
              {
                "type": "string",
                "value": "Total Loaded:"
              },
              {
                "type": "latest",
                "value": "jvm/classes/total_loaded"
              }
            ],
            [
              {
                "type": "string",
                "value": "Total Unloaded:"
              },
              {
                "type": "latest",
                "value": "jvm/classes/total_unloaded"
              }
            ]
          ],
          "timeseries": [
            {
              "type": "value",
              "attribute": "jvm/classes/current_loaded",
              "label": "# of currently loaded JVM Classes"
            }
          ]
        }
      }
    ]
  },
  "finagle": {
    "name": "Finagle",
    "runtime": "JVM",
    "summaryCard": {
      "lineOne": [
        {
          "type": "latest",
          "value": "finagle/future_pool/active_tasks"
        },
        {
          "type": "string",
          "value": "Active"
        }
      ],
      "lineTwo": [
        {
          "type": "latest",
          "value": "finagle/timer/pending_tasks.count"
        },
        {
          "type": "string",
          "value": "Pending"
        }
      ]
    },
    "grid": {
      "breakpoints": {
        "lg": 1200,
        "md": 996,
        "sm": 768
      },
      "cols": {
        "lg": 12,
        "md": 8,
        "sm": 4
      },
      "layouts": {
        "lg": [
          { "i": "Timer Deviation",     "x": 0, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Pending Timer Tasks", "x": 4, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Future Pool",         "x": 8, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Client Registry",     "x": 0, "y": 5, "w": 4, "h": 5, "minW": 3, "minH": 5 }
        ],
        "md": [
          { "i": "Timer Deviation",     "x": 0, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Pending Timer Tasks", "x": 4, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Future Pool",         "x": 8, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Client Registry",     "x": 0, "y": 5, "w": 4, "h": 5, "minW": 3, "minH": 5 }
        ],
        "sm": [
          { "i": "Timer Deviation",     "x": 0, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Pending Timer Tasks", "x": 4, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Future Pool",         "x": 8, "y": 0, "w": 4, "h": 5, "minW": 3, "minH": 5 },
          { "i": "Client Registry",     "x": 0, "y": 5, "w": 4, "h": 5, "minW": 3, "minH": 5 }
        ]
      },
      "rowHeight": 60
    },
    "charts": [
      {
        "title": "Timer Deviation",
        "type": "GMBasicMetrics",
        "data": {
          "detailLines": {
            "Count": "finagle/timer/deviation_ms.count",
            "Average": "finagle/timer/deviation_ms.avg",
            "Max": "finagle/timer/deviation_ms.max",
            "Min": "finagle/timer/deviation_ms.min",
            "Sum": "finagle/timer/deviation_ms.sum"
          }
        }
      },
      {
        "title": "Pending Timer Tasks",
        "type": "GMBasicMetrics",
        "data": {
          "detailLines": {
            "Count": "finagle/timer/pending_tasks.count",
            "Average": "finagle/timer/pending_tasks.avg",
            "Max": "finagle/timer/pending_tasks.max",
            "Min": "finagle/timer/pending_tasks.min",
            "Sum": "finagle/timer/pending_tasks.sum"
          }
        }
      },
      {
        "title": "Future Pool",
        "type": "GMBasicMetrics",
        "data": {
          "detailLines": {
            "Active Tasks": "finagle/future_pool/active_tasks",
            "Completed Tasks": "finagle/futurePool/completed_tasks",
            "Pool Size": "finagle/timer/pool_size"
          }
        }
      },
      {
        "title": "Client Registry",
        "type": "GMBasicMetrics",
        "data": {
          "detailLines": {
            "Initial Resolution": "finagle/clientregistry/initialresolution_ms",
            "Size": "finagle/clientregistry/size"
          }
        }
      }
    ]
  }
}
