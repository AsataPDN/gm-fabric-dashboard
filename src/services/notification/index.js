import { notification } from "uikit";

export function reportError(errorLabel, shouldTimeout, errorObject = "") {
  notification(errorLabel, {
    status: "danger",
    timeout: shouldTimeout ? 5000 : 0
  });
  console.log(errorLabel, errorObject);
}
