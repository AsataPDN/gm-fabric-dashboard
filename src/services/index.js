import "./instance";
import "./dashboards";
import "./fabricMicroservices";
