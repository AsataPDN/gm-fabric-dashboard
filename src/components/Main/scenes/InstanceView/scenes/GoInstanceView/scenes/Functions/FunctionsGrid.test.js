import React from "react";
import { mount, shallow } from "enzyme";
import configureStore from "redux-mock-store";
import _ from "lodash";

import * as noFuncState from "json/mockReduxState";
import * as state from "json/mockReduxStateMetrics";
import FunctionsGrid from "./index";
import NotFoundError from "components/Main/components/NotFoundError";
//import Action effects
import "services";
import Table from "components/Main/components/Table";

const mockStore = configureStore();
const mockState = state.default,
  noMetricsState = noFuncState.default;

// Filter out metrics starting with the key function
noMetricsState.instance.metrics = _.omitBy(
  noFuncState.default.instance.metrics,
  (value, key) => key.substr(0, 8) === "function"
);
let wrapper, FunctionsGridInstance;

const FunctionsGridWithMockStore = (
  <FunctionsGrid store={mockStore(mockState)} />
);
const FunctionsGridWithMissingMetricsStore = (
  <FunctionsGrid store={mockStore(noMetricsState)} />
);
let funcArr = [
  {
    func: "OrderItem",
    requests: 45313
  },
  {
    func: "ZZItem",
    requests: 323
  },
  {
    func: "AAStream",
    requests: 13
  }
];
const sortByOptions = [
  {
    value: "func",
    label: "Function"
  },
  {
    value: "requests",
    label: "Requests"
  },
  {
    value: "errorCount",
    label: "Error %"
  },
  {
    value: "latency50",
    label: "Latency 50%"
  },
  {
    value: "latency99",
    label: "Latency 99%"
  }
];

describe("Go Instance Functions View: <FunctionsGrid/>", () => {
  test("Matches snapshot", () => {
    const tree = shallow(FunctionsGridWithMockStore);
    expect(tree).toMatchSnapshot();
  });

  // FunctionsGridWithMissingMetricsStore does not contain any functions data
  test("returns NotFoundError if no functions are found ", () => {
    wrapper = mount(FunctionsGridWithMissingMetricsStore);

    expect(wrapper.find(NotFoundError).length).toBe(1);
  });

  test("returns correct number of <Table> and does not render <NotFoundError> when functions are found ", () => {
    wrapper = mount(FunctionsGridWithMockStore);
    expect(wrapper.find(Table).length).toBe(1);
    expect(wrapper.find(NotFoundError).length).toBe(0);
  });
});

describe("FunctionsGrid Child Components", () => {
  beforeEach(() => {
    wrapper = mount(FunctionsGridWithMockStore);
    FunctionsGridInstance = wrapper.find("FunctionsGrid").instance();
  });

  test("passes props to FunctionsTableToolbar", () => {
    FunctionsGridInstance = wrapper.find("FunctionsGrid").instance();
    expect(wrapper.find("TableToolbar").props()).toMatchObject({
      searchInputProps: {
        filterString: "",
        setFilterString: FunctionsGridInstance.setFilterString,
        searchPlaceholder: "Search Functions"
      },
      sortByProps: {
        sortByOptions,
        sortByAttribute: "func",
        setSortByAttribute: FunctionsGridInstance.setKeyToSortBy
      }
    });
  });

  test("passes functions as props to Table", () => {
    expect(wrapper.find(Table).props()).toMatchObject({
      items: [
        {
          errorPercent: "0.000",
          errorsCount: 0,
          func: "CatalogStream",
          inThroughput: 227,
          latency50: 1956,
          latency90: 3469,
          latency95: 3484,
          latency99: 3484,
          latency9990: 3484,
          latency9999: 3484,
          latencyAvg: 1942.692308,
          latencyCount: 13,
          latencyMax: 3484,
          latencyMin: 600,
          outThroughput: 2889,
          requests: 13
        },
        {
          errorPercent: "0.000",
          errorsCount: 0,
          func: "OrderItem",
          inThroughput: 225,
          latency50: 2339,
          latency90: 3476,
          latency95: 3565,
          latency99: 3565,
          latency9990: 3565,
          latency9999: 3565,
          latencyAvg: 2398.615385,
          latencyCount: 13,
          latencyMax: 3565,
          latencyMin: 1345,
          outThroughput: 143,
          requests: 13
        }
      ]
    });
  });
});

describe("FunctionsGrid Instance Methods", () => {
  beforeEach(() => {
    FunctionsGridInstance = wrapper.find("FunctionsGrid").instance();
  });
  test("`sort` takes functions and sorts by `funcs` in ascending order by default when `keyToSortBy` is not set and `ascending` is not yet toggled", () => {
    expect(FunctionsGridInstance.sort(funcArr)).toEqual([
      { func: "AAStream", requests: 13 },
      { func: "OrderItem", requests: 45313 },
      { func: "ZZItem", requests: 323 }
    ]);
  });

  test("`sort` sorts by `keyToSortBy` and `ascending` values", () => {
    FunctionsGridInstance.setState({
      keyToSortBy: "requests",
      ascending: false
    });
    expect(FunctionsGridInstance.sort(funcArr)).toEqual([
      { func: "OrderItem", requests: 45313 },
      { func: "ZZItem", requests: 323 },
      { func: "AAStream", requests: 13 }
    ]);
  });

  test("`setKeyToSortBy` sets keyToSortBy and toggles ascending order when key is already active ", () => {
    FunctionsGridInstance.setState({
      keyToSortBy: "requests",
      ascending: false
    });

    expect(FunctionsGridInstance.sort(funcArr)).toEqual([
      { func: "OrderItem", requests: 45313 },
      { func: "ZZItem", requests: 323 },
      { func: "AAStream", requests: 13 }
    ]);

    // same key is sent, which toggles ascending
    FunctionsGridInstance.setKeyToSortBy("requests");

    expect(FunctionsGridInstance.sort(funcArr)).toEqual([
      { func: "AAStream", requests: 13 },
      { func: "ZZItem", requests: 323 },
      { func: "OrderItem", requests: 45313 }
    ]);
  });
  test("`setKeyToSortBy` sorts by keyToSortBy ", () => {
    FunctionsGridInstance.setState({
      keyToSortBy: "func"
    });

    expect(FunctionsGridInstance.sort(funcArr)).toEqual([
      { func: "AAStream", requests: 13 },
      { func: "OrderItem", requests: 45313 },
      { func: "ZZItem", requests: 323 }
    ]);
  });

  test("`setFilterString` sets filterString ", () => {
    FunctionsGridInstance.setFilterString("AA");
    expect(FunctionsGridInstance.state.filterString).toEqual("AA");
    FunctionsGridInstance.setFilterString("Cat");
    expect(FunctionsGridInstance.state.filterString).toEqual("Cat");
  });
});
