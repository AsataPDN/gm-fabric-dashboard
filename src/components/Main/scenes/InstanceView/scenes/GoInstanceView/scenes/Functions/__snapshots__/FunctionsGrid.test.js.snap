// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Go Instance Functions View: <FunctionsGrid/> Matches snapshot 1`] = `
<div>
  <div
    className="sc-ckVGcZ konKeT"
  >
    <div
      className="sc-jKJlTe hUnWQY"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          aria-label="Search Functions"
          className="sc-kGXeez ljbeeX"
          onChange={[Function]}
          placeholder="Search Functions"
          type="search"
          value=""
        />
      </form>
    </div>
    <div
      className="sc-hMqMXs jBhXSB"
    />
    <div
      className="sc-eNQAEJ ifrloj"
    >
      <div
        className="Select sc-kpOJdX kDtvQI Select--single has-value"
        style={undefined}
      >
        <input
          disabled={false}
          name="form-field-sort-by"
          type="hidden"
          value="func"
        />
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-value"
              style={undefined}
              title={undefined}
            >
              <span
                aria-selected="true"
                className="Select-value-label"
                id="react-select-2--value-item"
                role="option"
              >
                <span>
                  <span>
                    Sort
                     
                  </span>
                  <span
                    className="sc-dxgOiQ idZMDV"
                  >
                    Function
                  </span>
                </span>
              </span>
            </div>
            <div
              aria-activedescendant="react-select-2--value"
              aria-expanded={false}
              aria-owns=""
              aria-readonly="false"
              className="Select-input"
              onBlur={[Function]}
              onFocus={[Function]}
              role="combobox"
              style={
                Object {
                  "border": 0,
                  "display": "inline-block",
                  "width": 1,
                }
              }
              tabIndex={0}
            />
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-cSHVUG YOnkJ"
  >
    <div
      className="sc-kAzzGY dYUejF"
    >
      <div
        className="sc-chPdSV HHZwB"
        style={
          Object {
            "flex": "1 1 20%",
          }
        }
      >
        Function
      </div>
      <div
        className="sc-chPdSV HHZwB"
      >
        Requests/sec
      </div>
      <div
        className="sc-chPdSV HHZwB"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Requests
      </div>
      <div
        className="sc-chPdSV HHZwB"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Error %
      </div>
      <div
        className="sc-chPdSV HHZwB"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Latency 50%
      </div>
      <div
        className="sc-chPdSV HHZwB"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Latency 99%
      </div>
    </div>
    <ol
      className="sc-kgoBCf dHvlms"
    >
      <li
        className="sc-iwsKbI geNuyD"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <div
          className="sc-gqjmRU gbBQLn"
        >
          CatalogStream
          <div
            className="sc-fjdhpX enphkh"
          >
            <div
              className="sc-jzJRlG hdTTKB"
              width={100}
            />
          </div>
        </div>
        <div
          className="sc-VigVT iaXEFZ"
        >
          <svg
            preserveAspectRatio="xMaxYMin"
            style={undefined}
            viewBox="0 0 240 32"
          >
            <g>
              <circle
                cx={2}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={61}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={120}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={179}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={238}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <polyline
                points="2 16 61 16 120 16 179 16 238 16 238 30 2 30 2 16"
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <polyline
                points="2 16 61 16 120 16 179 16 238 16"
                style={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": 1,
                  }
                }
              />
            </g>
            <line
              style={
                Object {
                  "opacity": "0.4",
                  "stroke": "grey",
                }
              }
              x1={2}
              x2={238}
              y1={18}
              y2={18}
            />
          </svg>
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          13
        </div>
        <div
          className="sc-gZMcBi deEsMZ"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          0.000%
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          1956
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          3484
        </div>
      </li>
      <li
        className="sc-iwsKbI geNuyD"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <div
          className="sc-gqjmRU gbBQLn"
        >
          OrderItem
          <div
            className="sc-fjdhpX enphkh"
          >
            <div
              className="sc-jzJRlG hdTTKB"
              width={100}
            />
          </div>
        </div>
        <div
          className="sc-VigVT iaXEFZ"
        >
          <svg
            preserveAspectRatio="xMaxYMin"
            style={undefined}
            viewBox="0 0 240 32"
          >
            <g>
              <circle
                cx={2}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={80.66666666666667}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={159.33333333333334}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <circle
                cx={238}
                cy={16}
                onClick={[Function]}
                onMouseEnter={[Function]}
                r={2}
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <polyline
                points="2 16 80.66666666666667 16 159.33333333333334 16 238 16 238 30 2 30 2 16"
                style={
                  Object {
                    "fill": "none",
                    "fillOpacity": ".1",
                    "pointerEvents": "auto",
                    "stroke": "currentColor",
                    "strokeWidth": "0",
                  }
                }
              />
              <polyline
                points="2 16 80.66666666666667 16 159.33333333333334 16 238 16"
                style={
                  Object {
                    "fill": "none",
                    "stroke": "currentColor",
                    "strokeLinecap": "round",
                    "strokeLinejoin": "round",
                    "strokeWidth": 1,
                  }
                }
              />
            </g>
            <line
              style={
                Object {
                  "opacity": "0.4",
                  "stroke": "grey",
                }
              }
              x1={2}
              x2={238}
              y1={18}
              y2={18}
            />
          </svg>
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          13
        </div>
        <div
          className="sc-gZMcBi deEsMZ"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          0.000%
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          2339
        </div>
        <div
          className="sc-gZMcBi dKkxkg"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          3565
        </div>
      </li>
    </ol>
  </div>
</div>
`;
